### ISSUE TITLE
definition of "unread count" in push gateway API does not match reality

### ISSUE THREAD

**Issue by richvdh** [matrix-org/matrix-doc#2627]
The [`_matrix/push/v1/notify`
endpoint](https://matrix.org/docs/spec/push_gateway/r0.1.1#post-matrix-push-v1-notify)
includes a parameter `notification.counts.unread`, defined as "The number of
unread messages a user has across all of the rooms they are a member of."

Since https://github.com/matrix-org/synapse/pull/793, Synapse has chosen to interpret this as "the number of rooms with unread messages". 

Possibly the spec should be reworded to allow discretion on the part of the homeserver impl?

**Comment by richvdh** [comment]
according to @BillCarsonFr , the riot mobile clients don't even use this field any more. maybe we can just drop it?

**Comment by richvdh** [comment]
while we're here: the spec also has `notification.counts.missed_calls`, which is fairly handwavily defined, and afaict has never been implemented by synapse

**Comment by richvdh** [comment]
but according to @ismailgulek, riot-iOS *does* use this field.

**Comment by ismailgulek** [comment]
> but according to @ismailgulek, riot-iOS _does_ use this field.

Yes, i confirm Riot-iOS will need `notification.counts.unread` in the next App Store version, but not `notification.counts.missed_calls`

**Comment by richvdh** [comment]
> but according to @ismailgulek, riot-iOS does use this field.

but we could still consider getting rid of `notification.counts.missed_calls`, which is unimplemented.

**Comment by richvdh** [comment]
it's now clear that element-iOS *does* use `notification.counts.unread`, whereas element-android does not. it seems like we should add a way for clients to opt in or out of receiving this field. If they have no purpose for it, there is no benefit of the HS and pushgw spending CPU cycles sending pushes, and the clients spending battery receiving them, only for them to be ignored.

**Comment by callahad** [comment]
Note that as of https://github.com/matrix-org/synapse/pull/8820 in Synapse 1.24, it's _possible_ (but not the default) to configure Synapse to meet the Spec's original definition of "The number of unread messages a user has across all of the rooms they are a member of."
